shared/fs/footer/upload.js.flow

```diff
@@ -4,9 +4,10 @@ import * as React from 'react'
 export type UploadProps = {
   showing: boolean,
   files: number,
+  fileName: string | null,
```

Comment:
Not that you have to change, but I think this could also be `fileName: ?string`, which is equivalent to what you have. Another form you might see in the codebase is `fileName?: string`, or `fileName?: ?string`. The difference is, if the `?` is after the field name, it means the field is optional, i.e. can either be the type indicated, or `undefined`. If the `?` is right before the type, it means the field has to present, but can either be the type indicated, or `null`. Similarly, `fileName?: ?string` means `fileName` can be either a string, `null`, or `undefined`.

Reply:
Thanks, let me resolve this and the comments above

Reply:
Under @taruti 's prompt I'm realizing that I was being inaccurate here when I said `fileName: ?string` was equivalent to `fileName: string | null`. `fileName: ?string` actually covers both `null` and `undefined`. `fileName?` indicates that it's optional. Please see [Maybe types](https://flow.org/en/docs/types/primitives/#toc-maybe-types) and [Optional](https://flow.org/en/docs/types/primitives/#toc-optional-object-properties) stuff in the Flow doc.

Not suggesting any change here, but since I gave mis-information here I wanted to make sure to correct it. Sorry about that!